lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -89,8 +104,12 @@ public void collect(int doc) throws IOException {
           pqTop.score = score;
           pqTop = pq.updateTop();
           updateMinCompetitiveScore(scorer);
-        }
 
+          if (bottomValueChecker != null && bottomValueChecker.isGlobalBottomValueAvailable()) {
```

Comment:
Same here, this should move `updateMinCompetitiveScore`, as is it doesn't respect the total hits threshold and the  bottom values might be greater than the local bottom score.